mountpoint-s3-client/src/object_client.rs

```diff
@@ -264,6 +272,24 @@ pub enum DeleteObjectError {
     NoSuchBucket,
 }
 
+/// Result of a [`copy_object`](ObjectClient::copy_object) request
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct CopyObjectResult {
+    // TODO: Populate this struct with return fields from the S3 API, e.g., etag.
```

Comment:
Do we want to keep this as a `TODO`, or shall we implement _some_ of the return fields?

Reply:
This is an issue from CRT where it does not send us any response body. Once they fix, we will be able to implement.